### PR TITLE
Add --lint-fix option.

### PR DESCRIPTION
**Pull request by wprl**
This PR writes any fixes eslint detects to disk if `--lint-fix` is used.

### CODE REVIEW

**Comment by wprl** [comment]
Tests pass now!

File: lib/linter/index.js [hunk 1]
```diff
@@ -42,17 +42,29 @@ exports.lint = function () {
...
-            filename: result.filePath,
-            errors: result.messages.map((err) => {
+        const transformed = {
+            filename: result.filePath
+        };
+
+        if (result.hasOwnProperty('output')) {
+
+            transformed.fix = {
+                filePath: result.filePath,
```
Comment (geek) [inline lib/linter/index.js]: shouldn't need `filePath` set here, can use the `filename` property of `transformed`

File: lib/lint.js [hunk 2]
```diff
@@ -29,20 +42,27 @@ exports.lint = function (settings, callback) {
...
+                });
+            }
 
             lint.totalErrors = errors;
             lint.totalWarnings = warnings;
             result.totalErrors += errors;
             result.totalWarnings += warnings;
+
+            if (lint.fix) {
+                Fs.writeFileSync(lint.fix.filePath, lint.fix.output);
```
Comment (geek) [inline lib/lint.js]: can use `lint.filename`

File: lib/lint.js [hunk 3]
```diff
@@ -14,10 +15,22 @@ const internals = {
 
 exports.lint = function (settings, callback) {
 
+    const linterOptions = settings['lint-options'] ? [settings['lint-options']] : ['{}'];
```
Comment (geek) [inline lib/lint.js]: Can this be changed to something like

```js
let linterOptions = JSON.parse(settings['lint-options'] || '{}');
linterOptions = Object.assign(linterOptions, { fix: !!settings['lint-fix'] });
const linterPath = (settings.linter && settings.linter !== 'eslint') ? settings.linter : internals.linter;

const child = ChildProcess.fork(linterPath, [JSON.stringify(linterOptions)], { cwd: settings.lintingPath });
```

**Comment by geek** [comment]
@wprl you will need to update the cli definition with the new flag https://github.com/hapijs/lab/blob/71b927b57258a265107a62ba46a28053fb308785/lib/cli.js#L134  Please also indicate in the readme that the default is to not fix files.

**Comment by wprl** [comment]
How's this look?

File: lib/lint.js [hunk 4]
```diff
@@ -15,9 +16,21 @@ const internals = {
...
-                                  settings['lint-options'] ? [settings['lint-options']] : [],
-                                  { cwd: settings.lintingPath });
+
+    let linterOptions;
+
+    try {
+        linterOptions = JSON.parse(settings['lint-options']);
+    }
+    catch (err) {
+        linterOptions = {};
```
Comment (geek) [inline lib/lint.js]: In this case we should throw with an error informing the user that `lint-options` couldn't be parsed.

File: lib/lint.js [hunk 5]
```diff
@@ -15,9 +16,21 @@ const internals = {
...
+    }
+
+    if (settings['lint-fix']) {
+        linterOptions.fix = true;
```
Comment (geek) [inline lib/lint.js]: can assign directly now that lint-fix is always a boolean:

```js
linterOptions.fix = settings['lint-fix'];
```

**Comment by wprl** [comment]
OK see how you like this revision :)

**Pull request merged**